includes/REST_API_Factory.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Class REST_API_Factory.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use Google\Web_Stories\REST_API\Embed_Controller;
+use Google\Web_Stories\REST_API\Status_Check_Controller;
+use Google\Web_Stories\REST_API\Link_Controller;
+use Google\Web_Stories\REST_API\Stories_Autosaves_Controller;
+use Google\Web_Stories\REST_API\Stories_Lock_Controller;
+use Google\Web_Stories\REST_API\Stories_Media_Controller;
+use Google\Web_Stories\REST_API\Stories_Settings_Controller;
+use Google\Web_Stories\REST_API\Stories_Users_Controller;
+
+/**
+ * Class REST_API_Factory
+ */
+class REST_API_Factory extends Service {
+	/**
+	 * Registers REST API routes.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return void
+	 */
+	public function register() {
+
+		$link_controller = new Link_Controller();
+		$link_controller->register_routes();
+
+		$status_check = new Status_Check_Controller();
+		$status_check->register_routes();
+
+		$embed_controller = new Embed_Controller();
+		$embed_controller->register_routes();
+
+		$templates_autosaves = new Stories_Autosaves_Controller( Template_Post_Type::POST_TYPE_SLUG );
+		$templates_autosaves->register_routes();
+
+		$stories_autosaves = new Stories_Autosaves_Controller( Story_Post_Type::POST_TYPE_SLUG );
+		$stories_autosaves->register_routes();
+
+		$templates_lock = new Stories_Lock_Controller( Template_Post_Type::POST_TYPE_SLUG );
+		$templates_lock->register_routes();
+
+		$stories_lock = new Stories_Lock_Controller( Story_Post_Type::POST_TYPE_SLUG );
+		$stories_lock->register_routes();
+
+		$stories_media = new Stories_Media_Controller( 'attachment' );
+		$stories_media->register_routes();
+
+		$stories_users = new Stories_Users_Controller();
+		$stories_users->register_routes();
```

Comment:
These could also be turned into a series of service. The `register_routes()` method could just as well be a simple `register()`.

The constructor arguments for some of these can be set via (overriding) `ServiceBasedPlugin::get_arguments()`.

Reply:
Yeah, I thought this, but for now, I think I am going to leave it as it is and add a todo to the code.